server/src/main/java/org/elasticsearch/indices/SystemIndices.java

```diff
@@ -625,81 +624,8 @@ public static Feature pluginToFeature(SystemIndexPlugin plugin, Settings setting
-            ThreadPool.Names.SYSTEM_READ, ThreadPool.Names.SYSTEM_READ, ThreadPool.Names.SYSTEM_WRITE
-        );
-
-        /**
-         * The thread pools for a typical system data stream. These are also the usual thread
-         * pools for non-system indices and data streams.
-         */
-        public static DescriptorThreadPoolNames DEFAULT_SYSTEM_DATA_STREAM_THREAD_POOLS = new DescriptorThreadPoolNames(
-            ThreadPool.Names.GET, ThreadPool.Names.SEARCH, ThreadPool.Names.WRITE
-        );
-
-        /**
-         * The thread pools that should be used for critical system index operations.
-         */
-        public static DescriptorThreadPoolNames CRITICAL_SYSTEM_INDEX_THREAD_POOLS = new DescriptorThreadPoolNames(
-            ThreadPool.Names.SYSTEM_CRITICAL_READ, ThreadPool.Names.SYSTEM_CRITICAL_READ, ThreadPool.Names.SYSTEM_CRITICAL_WRITE
-        );
-
-        /**
-         * Create a new collection of thread pool names for a system descriptor to use.
-         * @param getPoolName Name of the thread pool that get operations should use.
-         * @param searchPoolName Name of the thread pool that search operations should use.
-         *                       (In same cases, this is the same as the name of the pool for
-         *                       get operations.)
-         * @param writePoolName Name of the thread pool that write operations should use.
-         */
-        public DescriptorThreadPoolNames(String getPoolName, String searchPoolName, String writePoolName) {
-            if (ThreadPool.THREAD_POOL_TYPES.containsKey(getPoolName) == false) {
-                throw new IllegalArgumentException(getPoolName + " is not a valid thread pool");
-            }
-            if (ThreadPool.THREAD_POOL_TYPES.containsKey(searchPoolName) == false) {
-                throw new IllegalArgumentException(searchPoolName + " is not a valid thread pool");
-            }
-            if (ThreadPool.THREAD_POOL_TYPES.containsKey(writePoolName) == false) {
-                throw new IllegalArgumentException(writePoolName + " is not a valid thread pool");
-            }
-            this.getPoolName = getPoolName;
-            this.searchPoolName = searchPoolName;
-            this.writePoolName = writePoolName;
-        }
-
-        /**
-         * @return Name of the thread pool that get operations should use
-         */
-        public String getGetPoolName() {
-            return getPoolName;
-        }
-
-        /**
-         * @return Name of the thread pool that search operations should use
-         */
-        public String getSearchPoolName() {
-            return searchPoolName;
-        }
-
-        /**
-         * @return Name of the thread pool that write operations should use
-         */
-        public String getWritePoolName() {
-            return writePoolName;
-        }
+    public ExecutorSelector getExecutorSelector() {
+        return new ExecutorSelector(this);
```

Comment:
do we need a new one every time? Couldn't we just have single thread safe instance?

Reply:
Yes, we should have a single thread safe instance. Mistake on my part, and I will fix it.